### PR TITLE
Assignment3: David & Oscar

### PR DESCRIPTION
**Pull request by Oscar-Ruelas**
Collab w/ @dav1dmoore

### CODE REVIEW

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 1]
```diff
@@ -1,7 +1,169 @@
+using System;
 using Xunit;
+using Interfaces;
+using Moq;
 
 namespace CanHazFunny.Tests;
+    public class JesterTests
+    {
+        [Fact]
```
Comment (tienv9) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:9]: There shouldn't be any spacing here. Try removing the semicolon and then adding it back to see if it fixes itself.

File: CanHazFunny/CanHazFunny/ConsoleJokeOutput.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+using Interfaces;
+using System;
+
+namespace CanHazFunny;
+
+
+    public class ConsoleJokeOutput : IJokeOutput
+    {
```
Comment (tienv9) [inline CanHazFunny/CanHazFunny/ConsoleJokeOutput.cs:8]: Same issue as above.

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+using Interfaces;
+using System;
+
+namespace CanHazFunny;
+
+    public class Jester
+    {
```
Comment (tienv9) [inline CanHazFunny/CanHazFunny/Jester.cs:7]: Same issue as above.

File: CanHazFunny/Interfaces/IJokeOutput.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+namespace Interfaces;
+public interface IJokeOutput
+{
+    void OutputJoke(string joke)
+    {
+        //Applying this interface to Jester to output to the console. No return so Void.
+    }
```
Comment (tienv9) [inline CanHazFunny/Interfaces/IJokeOutput.cs:7]: This can just be 
```suggestion
    void OutputJoke(string joke);
```

**Review by tienv9 — COMMENTED** [review]
Assignment
For this assignment do the following:

Create two interfaces. One to represent the JokeService and one interface to represent outputting the joke to the screen (Console.WriteLine). ✔
The JokeService will need to have the interface applied to it. ✔
Create an implementation of the output interface that writes out the joke to the console. ✔
Implement the Jester class. It should take in both interfaces as dependencies. These dependencies should be null checked. ✔
The Jester class TellJoke() method should retrieve a joke from the JokeService. If the joke contains "Chuck Norris", skip it and get another. The joke should be written to the output dependency. ✔
Unit test the Jester class. Code coverage should be above 90% for this class. Moq will make this significantly easier. ✔
Fundamentals
Be sure you enable:
Nullability for all projects ✔
Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
Ensure that you turn on code analysis for all projects(EnableNETAnalyzers) ✔
Ensure that you turn on treat warnings as errors for all projects ✔
Ensure that you turn on CodeAnalysisTreatWarningsAsErrors ✔
Ensure that you turn on EnforceCodeStyleInBuild ✔
For this assignment, always use Assert.Equal<T>() (the generic version) ✔
Ensure there are no errors or warnings (including code analysis warnings) ✔
All of the above should be unit tested ✔
Choose simplicity over complexity ✔
Extra Credit
Unit test your implementation that writes the joke out to the screen. How hard could it be to unit test a single line method ;)? ✔
The Geek jokes API that is being used can also return jokes in a JSON format. Update the JokeService to retrieve jokes using JSON. GetJoke should still return a string. ✔

**Comment by kflannery2001** [comment]
I will review this

**Comment by ccornell88** [comment]
Assignment
For this assignment do the following:

Create two interfaces. One to represent the JokeService and one interface to represent outputting the joke to the screen (Console.WriteLine). ✔
The JokeService will need to have the interface applied to it. ✔
Create an implementation of the output interface that writes out the joke to the console. ✔
Implement the Jester class. It should take in both interfaces as dependencies. These dependencies should be null checked. ✔
The Jester class TellJoke() method should retrieve a joke from the JokeService. If the joke contains "Chuck Norris", skip it and get another. The joke should be written to the output dependency. ✔
Unit test the Jester class. Code coverage should be above 90% for this class. Moq will make this significantly easier. ✔
Fundamentals
Be sure you enable:
Nullability for all projects ✔
Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
Ensure that you turn on code analysis for all projects(EnableNETAnalyzers) ✔
Ensure that you turn on treat warnings as errors for all projects ✔
Ensure that you turn on CodeAnalysisTreatWarningsAsErrors ✔
Ensure that you turn on EnforceCodeStyleInBuild ✔
For this assignment, always use Assert.Equal() (the generic version) ✔
Ensure there are no errors or warnings (including code analysis warnings) ✔
All of the above should be unit tested ✔
Choose simplicity over complexity ✔
Extra Credit
Unit test your implementation that writes the joke out to the screen. How hard could it be to unit test a single line method ;)?✔
The Geek jokes API that is being used can also return jokes in a JSON format. Update the JokeService to retrieve jokes using JSON. GetJoke should still return a string.✔

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs (hunk 1 above)
Comment (Oscar-Ruelas) [reply CanHazFunny/CanHazFunny.Tests/JesterTests.cs]: Oh I see what you mean. I have now fixed this

**Review by Oscar-Ruelas — COMMENTED** [review]
I resolved all the peer-review concerns.

File: CanHazFunny/Interfaces/IJokeOutput.cs (hunk 4 above)
Comment (Oscar-Ruelas) [reply CanHazFunny/Interfaces/IJokeOutput.cs]: I have fixed this

File: CanHazFunny/CanHazFunny/Jester.cs (hunk 3 above)
Comment (Oscar-Ruelas) [reply CanHazFunny/CanHazFunny/Jester.cs]: Fixed

File: CanHazFunny/CanHazFunny/ConsoleJokeOutput.cs (hunk 2 above)
Comment (Oscar-Ruelas) [reply CanHazFunny/CanHazFunny/ConsoleJokeOutput.cs]: Fixed

**Pull request closed without merging**